### PR TITLE
RSDK-6426 Update PTGs to output normalized velocities

### PR DESCRIPTION
**Pull request by biotinker**
This PR updates `motionplan/tpspace` such that PTGs no longer output specific velocities, but rather output normalized values between -1 and 1, which the receiver can use to adjust to any actual desired velocity.

We also remove all concept of `Time` from the PTG trajectories which are generated by tpspace, instead allowing the caller to compute their own timing based on their desired speeds.

This has been tested on real hardware and confirmed to work.

### CODE REVIEW

File: components/base/kinematicbase/ptgKinematics.go [hunk 1]
```diff
@@ -146,54 +173,58 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
...
+			// We need to update currentInputs as we move through the arc.
+			for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
+				distIncVel := step.linVelMMps.Y
+				if distIncVel == 0 {
+					distIncVel = step.angVelDegps.Z
+				}
+				ptgk.inputLock.Lock()
+				ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {math.Abs(distIncVel) * timeElapsed}}
+				ptgk.inputLock.Unlock()
+				utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
```
Comment (raybjork) [inline components/base/kinematicbase/ptgKinematics.go:224]: have the same nit here

File: components/base/kinematicbase/ptgKinematics.go [hunk 2]
```diff
@@ -146,54 +173,58 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
...
+			return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
 		}
+		utils.PanicCapturingGo(func() {
+			// We need to update currentInputs as we move through the arc.
+			for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
+				distIncVel := step.linVelMMps.Y
+				if distIncVel == 0 {
+					distIncVel = step.angVelDegps.Z
+				}
+				ptgk.inputLock.Lock()
```
Comment (raybjork) [inline components/base/kinematicbase/ptgKinematics.go:221]: Are we concerned about the rate at which we'll be switching this mutex?

File: components/base/kinematicbase/ptgKinematics.go [hunk 3]
```diff
@@ -247,6 +278,79 @@ func (ptgk *ptgBaseKinematics) ErrorState(ctx context.Context, plan [][]referenc
 	return spatialmath.PoseBetween(nominalPose, actualPIF), nil
 }
 
+func (ptgk *ptgBaseKinematics) trajectoryToArcsteps(traj []*tpspace.TrajNode) []arcStep {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply components/base/kinematicbase/ptgKinematics.go:281]: Also theres a world where adding these helpers to the tpspace package could make sense.  I'm fine with them being here though. wondering what your thoughts are?

File: components/base/kinematicbase/ptgKinematics.go [hunk 4]
```diff
@@ -247,6 +278,79 @@ func (ptgk *ptgBaseKinematics) ErrorState(ctx context.Context, plan [][]referenc
...
+		curDist += distIncrement
+		if nextStep.linVelMMps.Y != 0 {
+			timeStep += distIncrement / (math.Abs(nextStep.linVelMMps.Y))
+		} else if nextStep.angVelDegps.Z != 0 {
+			timeStep += distIncrement / (math.Abs(nextStep.angVelDegps.Z))
+		}
+	}
+	nextStep.timestepSeconds = timeStep
+	finalSteps = append(finalSteps, nextStep)
+	return finalSteps
```
Comment (raybjork) [inline components/base/kinematicbase/ptgKinematics.go:317]: I'm having a hard time reviewing this through inspection.  Have you tested this code on hardware?

**Review by raybjork — CHANGES_REQUESTED** [review]

File: components/base/kinematicbase/ptgKinematics.go [hunk 5]
```diff
@@ -146,54 +173,58 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
...
-				nil,
-			)
-			if err != nil {
-				stopCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*5)
-				defer cancelFn()
-				return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
-			}
-			lastLinVel = linVel
-			lastAngVel = angVel
+		timestep := time.Duration(step.timestepSeconds*1000*1000) * time.Microsecond
```
[the remark this answers is not in the document]
Comment (raybjork) [reply components/base/kinematicbase/ptgKinematics.go:194]: Oh I had the SI prefixes drilled into my head in middle school so the conversion is simpler when I'm looking at the order of magnitude.  If you did it intentionally and prefer this we can leave it though

**Review by raybjork — APPROVED** [review]

File: components/base/kinematicbase/ptgKinematics.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (biotinker) [reply components/base/kinematicbase/ptgKinematics.go:194]: Yeah I mentally count by multiples of a thousand. I really like how the `time` package does it
https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/time/time.go;l=631

**Pull request merged**